### PR TITLE
ci: update kubernetes build

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind ci
#### What this PR does / why we need it:
we should use the main branch, as well as set CRI stats in a less hacky way
#### Special notes for your reviewer:
more details in commits
#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by haircommander** [comment]
holy cow e2e_crun_cgroupv2 for the first time in months

File: contrib/test/integration/main.yml [hunk 1]
```diff
@@ -25,17 +25,17 @@
     - name: clone build and install kubernetes
       include: "build/kubernetes.yml"
       vars:
-        k8s_git_version: "release-1.21"
+        k8s_git_version: "master"
```
Comment (TomSweeneyRedHat) [inline contrib/test/integration/main.yml:28]: Are they really still using "master" and not "main"?  Sigh.

**Comment by TomSweeneyRedHat** [comment]
One sigh, otherwise LGTM

**Review by fgiudici — APPROVED** [review]
lgtm!

**Pull request closed without merging**